railties/lib/rails/test_help.rb

```diff
@@ -20,30 +20,35 @@
     exit 1
   end
 
-  module ActiveSupport
-    class TestCase
-      include ActiveRecord::TestDatabases
-      include ActiveRecord::TestFixtures
-      self.fixture_path = "#{Rails.root}/test/fixtures/"
-      self.file_fixture_path = fixture_path + "files"
-    end
+  ActiveSupport.on_load(:active_support_test_case) do
+    include ActiveRecord::TestDatabases
+    include ActiveRecord::TestFixtures
+
+    self.fixture_path = "#{Rails.root}/test/fixtures/"
+    self.file_fixture_path = fixture_path + "files"
   end
 
-  ActionDispatch::IntegrationTest.fixture_path = ActiveSupport::TestCase.fixture_path
+  ActiveSupport.on_load(:action_dispatch_integration_test) do
+    self.fixture_path = ActiveSupport::TestCase.fixture_path
+  end
 end
 
 # :enddoc:
 
-class ActionController::TestCase
-  def before_setup # :nodoc:
-    @routes = Rails.application.routes
-    super
+ActiveSupport.on_load(:action_controller_test_case) do
+  include Module.new do
```

Comment:
What benefit do we get from this anonymous module? The previous implementation was defining `before_setup` directly on the class, and the simplest change would be to continue to do that here inside the load hook.

(I can imagine at least one reason to use a module, but ideally it would be called out in the commit message!)